kerckhoff/packages/models.py

```diff
@@ -33,13 +33,34 @@ class Meta:
         unique_together = ('package_set', 'slug',)
 
 # Snapshot of a Package instance at a particular time
+# A PackageVersion object is a specific combination of PackageItem objects
 class PackageVersion(models.Model):
     id = models.UUIDField(primary_key=True, default=uuid.uuid4, editable=False)
-    package = models.ForeignKey(Package, on_delete=models.PROTECT)
-    creator = models.ForeignKey(settings.AUTH_USER_MODEL, on_delete=models.CASCADE)
+    package = models.ForeignKey(Package, on_delete=models.CASCADE)
+    creator = models.ForeignKey(settings.AUTH_USER_MODEL, on_delete=models.PROTECT)
     version_description = models.TextField(blank=True)
     created_at = models.DateTimeField(auto_now_add=True)
     updated_at = models.DateTimeField(auto_now=True)
-    
+ 
     #TODO 
     # Add package stateEnum for future (freeze should change state)
+
+class PackageItem(models.Model):   
+    TEXT = 'txt'
+    AML = 'aml'
+    IMAGE = 'img'
+    MARKDOWN = 'mdn'
+    SPREADSHEET = 'xls'
+    
+    DTYPE_CHOICES = (
+        (TEXT, "TEXT"),
+        (AML, "ARCHIEML"),
+        (IMAGE, "IMAGE"),
+        (MARKDOWN, "MARKDOWN"),
+        (SPREADSHEET, "SPREADSHEET"),
+    )
+
+    id = models.UUIDField(primary_key=True, default=uuid.uuid4, editable=False)
+    package_version = models.ForeignKey(PackageVersion, on_delete=models.CASCADE)
+    data_type = models.CharField(max_length=3, choices=DTYPE_CHOICES, default='text')
```

Comment:
the default value is wrong and could you also make the migrations?

Reply:
we might also want to add a filename field, and a mimetype field. The filename is just for convenience, and the mimetype helps us figure how to interpret the thing at a lower level (this could just be the google doc mimetypes - https://developers.google.com/drive/api/v3/mime-types). For mimetypes don't have to make this into choices, just a regular charfield should be fine